# U.S. Medical Insurance Costs
## Project Goals
I want to analyze the insurance costs of female participants to determine how BMI, age, and smoker status impact insurance cost. 

**Hypothesis:** Greater values for BMI and age, as well as an affirmative value for smoker status, are correlated with a higher insurance cost. 

## Step 1: Retrieve the Data

In [23]:
import csv

with open("insurance.csv") as insurance_csv:
    # Creating a list of dictionaries 
    insurance_data = list(csv.DictReader(insurance_csv))
        
    
#printing to verify contents
#for row in insurance_data:
#    print(row)

## Step 2: Find the Correlation Coefficients

In [24]:
import numpy as np

print("I will be calculating the correlation coefficient, r, for the following values, where r = -1 is a perfect negative correlation, r = 0 is no correlation, and r = 1 is a perfect positive correlation")
# ***BMI***
bmis = [float(row["bmi"]) for row in insurance_data]
charges = [float(row["charges"]) for row in insurance_data]

r = np.corrcoef(bmis, charges)[0, 1]
print("The correlation coefficient for BMI and insurance charges is: " + str(r))


# ***Age***
ages = [float(row["age"]) for row in insurance_data]
charges = [float(row["charges"]) for row in insurance_data]

r = np.corrcoef(ages, charges)[0, 1]
print("The correlation coefficient for age and insurance charges is: " + str(r))


# ***Smoker status***
smokers = [row["smoker"] for row in insurance_data]
charges = [float(row["charges"]) for row in insurance_data]

# Converting "yes" and "no" values in the smokers list to 1 and 0, respectively, in the smokers_num list 
smokers_num = []
for smoker in smokers:
    if (smoker == "yes"):
        smokers_num.append(1)
    elif (smoker == "no"):
        smokers_num.append(0)
    else:
        continue

r = np.corrcoef(smokers_num, charges)[0, 1]
print("The correlation coefficient for smoker status and insurance charges is: " + str(r))


I will be calculating the correlation coefficient, r, for the following values, where r = -1 is a perfect negative correlation, r = 0 is no correlation, and r = 1 is a perfect positive correlation
The correlation coefficient for BMI and insurance charges is: 0.19834096883362878
The correlation coefficient for age and insurance charges is: 0.2990081933306476
The correlation coefficient for smoker status and insurance charges is: 0.7872514304984782


## Step 3: Analyze Findings
Smoker status showed the highest correlation to insurance charge (r = 0.79), while BMI (r = 0.20) and age (r = 0.30) showed a weak correlation to insurance charge. This suggests that, of the three variables, smokers status is the best predictor of insurance cost. 